my-app/src/components/register/register.js

```diff
@@ -38,41 +71,50 @@ addUser = _ => {
   .then(response => response.json())
   .then(this.getUsers)
   .catch(err => console.error(err));
-
   this.props.history.push(`/login`);
   alert("User " + user.username + " was registered");
 }
 
-render(){
+render() {
   const {user} = this.state;
+  const {errors} = this.state;
   return (
       <div className="login-page">
           <div className="form">
+          <form onSubmit ={this.handleSubmit}>
+                {errors.map(error => (
+                <p key={error}>Error: {error}</p>))}
+              <h5>First Name</h5>
               <input 
               type="text"
-              placeholder="First Name" 
+              placeholder="Enter First Name"
               onChange={e => this.setState({user: {...user, firstname: e.target.value }})} 
               />
+              <h5>Last Name</h5>
               <input 
               type="text"
-              placeholder="Last Name" 
+              placeholder="Enter Last Name"
               onChange={e => this.setState({user: {...user, lastname: e.target.value }})} 
               />
+              <h5>Email</h5>
               <input 
               type="text"
```

Comment:
I believe you can make this input type = "email" which would help with validation.